helix-core/src/main/java/org/apache/helix/monitoring/mbeans/CustomizedViewMonitor.java

```diff
@@ -0,0 +1,151 @@
+  private final String _sensorName;
+  private HistogramDynamicMetric _updateToAggregationLatencyGauge;
+  public static final String UPDATE_TO_AGGREGATION_LATENCY_GAUGE =
+      "UpdateToAggregationLatencyGauge";
+
+  public CustomizedViewMonitor(String clusterName) {
+    _clusterName = clusterName;
+    _sensorName = String.format("%s.%s", MonitorDomainNames.CustomizedView.name(), _clusterName);
+    _updateToAggregationLatencyGauge =
+        new HistogramDynamicMetric(UPDATE_TO_AGGREGATION_LATENCY_GAUGE, new Histogram(
+            new SlidingTimeWindowArrayReservoir(getResetIntervalInMs(), TimeUnit.MILLISECONDS)));
+  }
+
+  @Override
+  public DynamicMBeanProvider register() throws JMException {
+    List<DynamicMetric<?, ?>> attributeList = new ArrayList<>();
+    attributeList.add(_updateToAggregationLatencyGauge);
+    doRegister(attributeList, MBEAN_DESCRIPTION, getMBeanName());
+    return this;
+  }
+
+  private ObjectName getMBeanName() throws MalformedObjectNameException {
+    return new ObjectName(String
+        .format("%s:%s=%s", MonitorDomainNames.CustomizedView.name(), "Cluster", _clusterName));
+  }
+
+  @Override
+  public String getSensorName() {
+    return _sensorName;
+  }
+
+  void recordUpdateToAggregationLatency(long latency) {
+    if (_updateToAggregationLatencyGauge != null) {
+      _updateToAggregationLatencyGauge.updateValue(latency);
+    }
+  }
+
+  /**
+   * Find updated customized states and report the aggregation latency of each customized state
+   * @param updatedCustomizedViews Customized views that have been updated, obtained from CustomizedStateOutput
+   * @param curCustomizedViews Current customized view values from the CustomizedViewCache
+   * @param updatedStartTimestamps All customized state START_TIME property values from CustomizedStateOutput
+   * @param updateSuccess If the customized view update to ZK is successful or not
+   * @param endTime The timestamp when the new customized view is updated to ZK
+   */
+  public void reportLatency(List<CustomizedView> updatedCustomizedViews,
+      Map<String, CustomizedView> curCustomizedViews,
+      Map<String, Map<Partition, Map<String, String>>> updatedStartTimestamps,
+      boolean[] updateSuccess, long endTime) {
+    if (updatedCustomizedViews == null || curCustomizedViews == null
+        || updatedStartTimestamps == null) {
+      LOG.warn("Cannot find updated time stamps for customized states, input parameter is null.");
+      return;
+    }
+
+    List<Long> collectedTimestamps = new ArrayList<>();
+
+    for (int i = 0; i < updatedCustomizedViews.size(); i++) {
+      if (!updateSuccess[i]) {
+        continue;
+      }
+      CustomizedView newCV = updatedCustomizedViews.get(i);
+      String resourceName = newCV.getResourceName();
+      CustomizedView oldCV =
+          curCustomizedViews.getOrDefault(resourceName, new CustomizedView(resourceName));
+      if (newCV.getRecord().equals(oldCV.getRecord())) {
```

Comment:
Is this still necessary? The updated customized views should have been compared and filtered.

Reply:
I was trying to be safe. But you are right, it's not necessary. Removed it.